ipie/utils/chunk_large_chol.py

```diff
@@ -7,7 +7,7 @@ def split_cholesky(ham_filename, nmembers):
     #
     with h5py.File(ham_filename, "r") as source_file:
         dataset = source_file["LXmn"][()]  # pylint: disable=no-member
-        num_chol = dataset.shape[-1]
+        num_chol = dataset.shape[0]
```

Comment:
Add a [unit test](https://github.com/JoonhoLee-Group/ipie/blob/develop/ipie/utils/tests/test_io.py#L29). I'm surprised no tests caught this. I would also encourage adding an integration test / MPI test.